Add Nerd Fonts support

iFreilicht opened the pull request:
See #17

All Nerd Fonts glyphs were parsed from the official cheat sheet and the .tsv look perfect and works.

I do still have an issue with some glyphs from "Material Design Icons" where some glyphs cause the whole line to be printed in right-to-left order, but that could be an issue with my setup/fonts. If anyone could try this out and see if it works on their system, that would be great :)

![img-2020-04-01-151933](https://user-images.githubusercontent.com/9742635/78141867-66c83000-742c-11ea-8389-903c65654200.png)

iFreilicht committed:
Add Nerd Fonts .tsv and importer

cspeterson committed:
Introduce new comprehensive emoji database. Add skin tone filtering

cspeterson committed:
README: update usage to include all options from help output

cspeterson commented:
@iFreilicht  I've tagged you in [a reply](https://github.com/cspeterson/splatmoji/pull/16#issuecomment-618046009) to a related pull request about how it would be best to handle "extra" databases like these

cspeterson commented:
Regarding the RTL renderings, I had a feeling that after spot-checking a couple of the entries checks out - the unicode codepoints that you and I are seeing rendered RTL are from scripts like Arabic that are *supposed* to render RTL.

That is, the glyphs are being overridden with your Nerd Font font, but Rofi still recognizes these are RTL codepoints and goes that way.

I have no idea off the top of my head whether that is something that can be configured away via Rofi's theming or options

cspeterson committed:
Fix slighty wrong but not-broken tsv spec in README

cspeterson committed:
README: emoticons update info (there are none)

iFreilicht commented:
Ah, that makes perfect sense! I'll have a look and see whether this can be remedied. I'll reply on the other ticket regarding additional databases.

ss1978 commented:
Also an option if splatmoji.config could refer to different urls, for the tsv files.
eg:
external_gitmoji=https://raw.githubusercontent.com/ss1978/splatmoji/master/data/gitmoji.tsv

cspeterson committed:
Complete refactor and add units tests

cspeterson committed:
Introduce build script for packaging deb and rpm

iFreilicht commented:
@ss1978 let's discuss this in #16 only. It's really confusing if the discussion is spread across multiple threads.

cspeterson committed:
LICENSE: whoops, include correct copyright line

cspeterson committed:
Strip unicode ltr mark from user selection

cspeterson committed:
Update emoticon database to prepent LTR marks to lines including RTL/A characters

so the lines are aligned correctly in Rofi

cspeterson committed:
Add verbose output flag to display some important specific values on stderr

cspeterson committed:
Include cli data files list in verbose output

cspeterson committed:
Build: set correct mode for executable before packaging

cspeterson committed:
Changelog: update for v1.1.0

cspeterson committed:
Fix package file permissions by explicitly setting them in a script after install

WPettersson committed:
WIP: Add history file support

This functions as best I can work out, but the configuration needs an
absolute location for the history file. The config file does not
interpret ${HOME} or ~, which might need discussion.

For issue #23

WPettersson committed:
Don't do things if nothing is selected

WPettersson committed:
Use ~ to refer to HOME

The history file will now default to ~/.local/state/splatmoji/history as
per XDG proposal for such things. The tilde is expanded to ${HOME} when
reading the config. This expansion will occur for every variable, as
long as the tilde occurs at the start of the variable value. I don't see
this causing any issues, and could be useful if you have either menu or
typing tools in ~/bin/

cspeterson committed:
Fix typo in help output for escapes

cspeterson committed:
Build: add grep as dependency. Whoops!

I may sometime eventually go back and do the grep aprts in bash, but meh

cspeterson committed:
Update README re new "recently used" functionality by @WPettersson

cspeterson committed:
Config file: do not specify history file by default. And, sort the file.

cspeterson committed:
History: shift around the handling of the history file

cspeterson committed:
Refactor tests to work with new history file code

cspeterson committed:
README: mention tests/testing finally

cspeterson committed:
Add test for `save_history` function

cspeterson committed:
In save_history, check file nonzero not just exists

cspeterson committed:
Shift escaping to go *after* the history is saved

cspeterson committed:
Fix `save_history` to handle a history length of 1, and make test try a range from 1 on up

cspeterson committed:
Fix `history_file` example in config file to not be broken! Whoops.

iFreilicht committed:
Change over to new database-format

iFreilicht commented:
I updated the PR to support the new database format.

iFreilicht committed:
Make nerdfont appear after emojis
